src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -174,32 +174,32 @@ internal sealed class DocumentAttributes : IChecksummedObject, IObjectWritable
             /// <summary>
             /// The Id of the document.
             /// </summary>
-            public DocumentId Id { get; }
+            public DocumentId Id { get; private set; }
```

Comment:
i'm not a huge fan of tihs.  this means we're only "by convention immutable" instead of being strictly immutable.   do we have anything demonstrating the savings here to make this worthwhile?

Reply:
We don't. I'm not saying we should do this. The PR was just to demonstrate the different patterns. 
The best approach would be to wait till we can use records.